### PR TITLE
#577 Perform dogfood check on Travis

### PR DESCRIPTION
**Pull request by mkordas**
For #577:

* added new build to execute freshly built Qulice over entire code
* due to Travis YML syntax limitations two different commands must be
 specified by environmental variables

### CODE REVIEW

**Comment by davvd** [comment]
@mkordas Thanks, I will find someone to review this PR soon

**Comment by davvd** [comment]
@krzyk review it please, it is a rather short pull request, no need to do a full code review

File: .travis.yml [hunk 1]
```diff
@@ -6,19 +6,25 @@ cache:
...
     - $HOME/.m2
 env:
-  global:
+  - matrix:
+    - COMMAND="mvn clean install -Pqulice --errors"
+    - COMMAND="mvn com.qulice:qulice-maven-plugin:1.0-SNAPSHOT:check -Dqulice.excludes='xml:.*'"
+  - global:
     - MAVEN_OPTS="-XX:MaxPermSize=2g -Xmx4g"
-    - JAVA_OPTS="-XX:MaxPermSize=2g -Xmx4g"
+    - JAVA_OPTS="-XX:MaxPermSizex=2g -Xmx4g"
```
Comment (krzyk) [inline .travis.yml]: @mkordas typo?

**Comment by krzyk** [comment]
@mkordas strange, there is no output printed from those commands

**Comment by mkordas** [comment]
@krzyk yeah, that's just early prototype, but I have no idea how to move it forward

**Comment by mkordas** [comment]
@krzyk exports get messed: https://travis-ci.org/teamed/qulice/jobs/104876504#L146
Do you have idea why?

**Comment by krzyk** [comment]
@mkordas unfortunately no, I would try to change order of `matrix` and `global`, and if that doesn't help then create an issue in travis-ci project, they are quite responsive

**Comment by mkordas** [comment]
@krzyk changing order didn't help, I've tried that already. I'll try with submitting issue to Travis.

**Comment by mkordas** [comment]
@krzyk silly mistake :)
```yml
env:
  - global:
    - ...
  - matrix:
    - ...
```
is not equivalent to

```yml
env:
  global:
    - ...
  matrix:
    - ...
```

Now it works fine. Are you ok with the approach and creating 2 more failing builds? If yes, you can assign me to the issue.

**Comment by krzyk** [comment]
@mkordas add also `--fail-never` to the dogfooded one and it will be slightly better, at least we will see all the checkstyle problems

**Comment by mkordas** [comment]
@krzyk done in fe57956

**Comment by krzyk** [comment]
@davvd this is not postponed

**Comment by davvd** [comment]
> @davvd this is not postponed

@krzyk thanks, I removed the "postponed" tag

**Comment by krzyk** [comment]
@davvd assign @mkordas

**Comment by krzyk** [comment]
@davvd assign @krzyk

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by mkordas** [comment]
@rultor deploy

**Comment by krzyk** [comment]
@rultor deploy

**Comment by davvd** [comment]
> @davvd assign @mkordas

@krzyk yes, done, @mkordas is at the task, please start! whoever worked with this task before, please stop immediately

**Comment by davvd** [comment]
> @davvd assign @krzyk

@krzyk sure! @krzyk the task is yours, please go ahead

**Comment by krzyk** [comment]
@davvd this is not postponed

**Comment by davvd** [comment]
> @davvd this is not postponed 

@krzyk right, I removed "postponed" tag

**Comment by davvd** [comment]
@krzyk Many thanks! **25 mins** were added to your account in Transaction ID `75881382` (task took zero)... extra minutes for review comments (c=10)... +25 added to your rating, at the moment it is: [+4578](http://www.netbout.com/b/35035?open=rating)

**Comment by davvd** [comment]
@rultor deploy now pls

**Pull request merged**